fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java

```diff
@@ -834,13 +831,15 @@ public TExecPlanFragmentParams plan(TUniqueId loadId, long txnId, String label)
         if (db == null) {
             throw new MetaNotFoundException("db " + dbId + " does not exist");
         }
+
+        Table table = db.getTable(this.tableId);
+        if (table == null) {
+            throw new MetaNotFoundException("table " + this.tableId + " does not exist");
+        }
+
         Locker locker = new Locker();
-        locker.lockDatabase(db, LockType.READ);
+        locker.lockTablesWithIntensiveDbLock(db, Lists.newArrayList(table.getId()), LockType.READ);
```

Comment:
Does `PulsarRoutineLoadJob` need to be changed also?

Reply:
No, kafka and Pulsar will share RoutineLoadJob